atomic_reactor/utils/koji.py

```diff
@@ -576,39 +589,38 @@ def get_repositories_and_digests(workflow, pullspec_image):
     """
     Returns a map of images to their repositories and a map of media types to each digest
 
-    it creates a map of images to digests, which is need to create the image->repository
+    it creates a map of images to digests, which is needed to create the image->repository
     map and uses the same loop structure as media_types->digest, but the image->digest
     map isn't needed after we have the image->repository map and can be discarded.
     """
     digests = {}  # image -> digests
     typed_digests = {}  # media_type -> digests
-    for registry in workflow.data.push_conf.docker_registries:
-        for image in workflow.data.tag_conf.images:
-            image_str = image.to_str()
-            if image_str in registry.digests:
-                image_digests = registry.digests[image_str]
-                digest_list = [select_digest(image_digests)]
-                digests[image.to_str(registry=False)] = digest_list
-                for digest_version in image_digests.content_type:
-                    if digest_version == 'v1':
-                        continue
-                    if digest_version not in image_digests:
-                        continue
-                    digest_type = get_manifest_media_type(digest_version)
-                    typed_digests[digest_type] = image_digests[digest_version]
-
-    registries = workflow.data.push_conf.all_registries
+
+    for image in workflow.data.tag_conf.images:
+        image_digests = get_manifest_digests(image, workflow.conf.registry['uri'],
+                                             workflow.conf.registry['insecure'],
+                                             workflow.conf.registry.get('secret', None))
+        if image_digests:
+            digest_list = [select_digest(image_digests)]
+            digests[image.to_str(registry=False)] = digest_list
+            for digest_version in image_digests.content_type:
+                if digest_version == 'v1':
+                    continue
+                if digest_version not in image_digests:
+                    continue
+                digest_type = get_manifest_media_type(digest_version)
+                typed_digests[digest_type] = image_digests[digest_version]
+
     repositories = []
-    for registry in registries:
-        image = pullspec_image.copy()
-        image.registry = registry.uri
-        pullspec = image.to_str()
 
-        repositories.append(pullspec)
+    image = pullspec_image.copy()
+    pullspec = image.to_str()
+
+    repositories.append(pullspec)
 
-        digest_list = digests.get(image.to_str(registry=False), ())
-        for digest in digest_list:
-            digest_pullspec = image.to_str(tag=False) + "@" + digest
-            repositories.append(digest_pullspec)
+    digest_list = digests.get(image.to_str(registry=False), ())
+    for digest in digest_list:
+        digest_pullspec = image.to_str(tag=False) + "@" + digest
+        repositories.append(digest_pullspec)
```

Comment:
There's no need to `copy()` the pullspec_image anymore (or even have the `image` variable at all)

Reply:
removed it